clang/lib/Sema/SemaDecl.cpp

```diff
@@ -6347,6 +6347,26 @@ NamedDecl *Sema::ActOnVariableDeclarator(
     return nullptr;
   }
 
+  if (R->isSamplerT()) {
```

Comment:
Shouldn't we guard this code under `if (getLangOpts().SYCL || getLangOpts().OpenCL)` ?

Reply:
But this code already was under `if (getLangOpts().OpenCL)`, maybe this if could help to understand this code in the future.

Reply:
@AnastasiaStulova asked for applying changes like this as removing redundant checks should speed-up compilation.